weedcoco/utils.py

```diff
@@ -50,3 +51,8 @@ def add_collection_from_file(coco, collection_path):
         for annotation in coco["annotations"]
     ]
     return coco
+
+
+def get_image_average_hash(path):
+    """Return an average hash of an image"""
+    return str(imagehash.average_hash(PIL.Image.open(path)))
```

Comment:
Interesting. I had thought we'd use a cryptographic hash – to avoid byte-identical images – rather than a locality-sensitive hash. I don't know if it will be overly sensitive, when this algorithm was, I think, designed for finding similar images, and for showing them to a person to confirm.

Reply:
It would be good to check that there are no false positive hash identities on RDS at the moment, or images that are very close in hash. That would mean that we should increase the hash size.

Reply:
There are [several ways](https://fullstackml.com/wavelet-image-hash-in-python-3504fdd282b5) to hash that library offers, though I think any of them should be sufficient as the identifier. If we want to make it more precise, we can combine some of these hashes, which will cost more time in processing as well.

Reply:
The different hashes should have different *types* of false positive, but the main way to reduce false positives for LSH is to increase the hash size.

Reply:
The default hashsize is 8. I increased it to 16 in the latest commit. A sample file name can be `0b5c1a7c18fc49f80dfc04fc0cfc41f901f800d82018009c019c09bc033c017c.png`